### PR TITLE
CI and Build Scripts

### PR DESCRIPTION
**Pull request by ChrisMaddock**
Last set of CI and Build Scripts. As before, scripts are mainly as discussed in nunit/nunit-vb-samples#4 and nunit/nunit-csharp-samples#6.

Changes:

- NUnit is now referenced to a version restored by NuGet, pulled by the bootstrapping scripts. (NuGet Restore on the C++ project doesn't seem to be an option.)

I've removed the Travis build here, as I was unable to get it up and running. The error I couldn't get past is shown in the build for the first commit, if anyone with more experience has any ideas - but as far as I can see, mono doesn't support building vcxproj's yet. I imagine we could have a separate Unix build outside of the vcxproj, but that seemed like overkill for a sample repo.

### CODE REVIEW

**Comment by rprouse** [comment]
I am pretty sure that mono cannot build C++/CLI. Unless someone says otherwise, I will delete the Travis build for C++.

**Comment by CharliePoole** [comment]
Right... it's not supported.

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+## Build Scripts
+
+Each of the sample solutions can be built individually within Visual Studio. To build and test all solutions, a **Cake** (http://cakebuild.net) script is provided.
+The primary script that controls this is build.cake. We modify build.cake when we need to add new 
+targets or change the way the build is done. Normally build.cake is not invoked directly but through
+build.ps1 (on Windows) or build.sh (on Linux). These two scripts are provided by the Cake project
```
Comment (CharliePoole) [inline BUILDING.md]: Remove reference to linux.

File: build.cake [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+.IsDependentOn("Clean")
+.IsDependentOn("Build");
+
+Task("Appveyor")
+.IsDependentOn("Build")
+.IsDependentOn("Test");
+
+Task("Travis")
+.IsDependentOn("Build")
+.IsDependentOn("Test");
```
Comment (CharliePoole) [inline build.cake]: Remove Travis task.

**Comment by CharliePoole** [comment]
I would say it can be merged once the remaining linux stuff is removed.

**Comment by rprouse** [comment]
I've disabled Travis builds for this project. I see that VB has never passed on Travis either. Should it be disabled there too?

**Comment by CharliePoole** [comment]
@rprouse I thought that VB was supposed to work under Mono... there might be something extra that we have to install.

**Comment by ChrisMaddock** [comment]
On VB, we hit a [compiler bug in VBNC regarding generics](https://github.com/nunit/nunit-vb-samples/pull/4#issuecomment-200360552), so we removed the tasks there. From the PR, it looks like the Travis CI is already disabled, @rprouse?

@CharliePoole - I left the unix references in Building.md/Cake to keep the docs standard between all repos - as we did with the VB repo. My thinking at the time was that we could then update all 4 sample repos in unison. 

I guess the cake script is already different in this repo, as it's already customised to remove the Nuget restore. Given 2/4 sample repos aren't on Unix now, I guess there's not much standardisation anyway - so I'll remove the references here and in the VB-samples in the morning.

**Comment by CharliePoole** [comment]
Thanks for the reminder... I see we already discussed this!

**Comment by rprouse** [comment]
@ChrisMaddock thanks for the update. I checked the Travis build and it was turned off in the settings, but still checking the repo. I removed it from the list of current builds on Travis since it isn't working and I don't like seeing red :smile: 

Ping us once you finish the cleanup and @CharliePoole or I will merge.

**Comment by ChrisMaddock** [comment]
@rprouse @CharliePoole - Unix references removed, apologies it took a few days! Will make the corresponding changes to the VB repo now as well.

**Comment by ChrisMaddock** [comment]
@rprouse - mind taking another look at this? It's good to go, from my point of view. Thanks in advance.

**Comment by rprouse** [comment]
Looks good and CI is passing.

**Comment by ChrisMaddock** [comment]
Cheers @rprouse!

**Pull request merged**